src/app/core/header/timepiece/timepiece.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Observable } from 'rxjs/Rx';
+import { Subscription } from 'rxjs';
+
+@Component({
+    selector: 'sh-timepiece',
+    styleUrls: ['./timepiece.component.scss'],
+    templateUrl: './timepiece.component.html'
+})
+
+export class TimepieceComponent implements OnInit, OnDestroy {
+
+    private currentTime: Date;
+    private timer: Observable<Object>;
+    private timeTracker: Subscription;
+
+    ngOnInit() {
+        this.createTimer(0, 1000);
```

Comment:
This changes should be merged till tomorrow. I'm going to implement your solution as a refactoring tech task